### ISSUE TITLE
Failed to determine at least one MIME-type

### ISSUE THREAD

**Issue by 875d** [ahrm/sioyek#529]
The `ss` and `sl` commands don't work for me:

```
Warning: invalid command (key: resetting to root
WARNING: failed to determine at least one MIME-type for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
WARNING: failed to determine command precursor for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
WARNING: failed to determine command(s) for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
Warning: invalid command (key:S); resetting to root
WARNING: failed to determine at least one MIME-type for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
WARNING: failed to determine command precursor for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
WARNING: failed to determine command(s) for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
WARNING: failed to determine at least one MIME-type for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
WARNING: failed to determine command precursor for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
WARNING: failed to determine command(s) for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
Warning: invalid command (key:S); resetting to root
Warning: invalid command (key:S); resetting to root
WARNING: failed to determine at least one MIME-type for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
WARNING: failed to determine command precursor for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
WARNING: failed to determine command(s) for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
Warning: invalid command (key:S); resetting to root
WARNING: failed to determine at least one MIME-type for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
WARNING: failed to determine command precursor for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
WARNING: failed to determine command(s) for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
Warning: invalid command (key:S); resetting to root
WARNING: failed to determine at least one MIME-type for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
WARNING: failed to determine command precursor for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
WARNING: failed to determine command(s) for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
Warning: invalid command (key:S); resetting to root
WARNING: failed to determine at least one MIME-type for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
WARNING: failed to determine command precursor for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
WARNING: failed to determine command(s) for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
Warning: invalid command (key:S); resetting to root
WARNING: failed to determine at least one MIME-type for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
WARNING: failed to determine command precursor for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
WARNING: failed to determine command(s) for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
Warning: invalid command (key:S); resetting to root
WARNING: failed to determine at least one MIME-type for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
WARNING: failed to determine command precursor for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
WARNING: failed to determine command(s) for Vaswani,%20Ashish,%20et%20al.%20%22Attention%20is%20all%20you%20need.%22%20Advances%20in%20neural%20information%20processing%20systems.%202017.%20
```

I looked for "failed to determine command(s)" in the source code but can't find it.

**Comment by ahrm** [comment]
Have you customized your external search URLs? If so, make sure that they start with `http://` or `https://`.

**Comment by 875d** [comment]
No, the default `prefs.config` was not loaded, so there were no search urls.

But sioyek should throw a more useful error (search key not defined) instead of trying to open the text with an empty search URL

**Comment by ahrm** [comment]
Fixed in e055a821e36a114deeb572c2bb665ee4af98599a.

**Comment by 875d** [comment]
Thanks! :)
